.github/workflows/checkmake.yaml

```diff
@@ -0,0 +1,35 @@
+name: ShellCheck
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+      - release-*
+  pull_request:
```

Comment:
How about if we only run this CI on pull requests and not on `push`?

Reply:
good idea! changed accordingly.